components/engine_traits/src/raft_engine.rs

```diff
@@ -90,6 +90,8 @@ pub trait RaftLogBatch: Send {
     fn put_raft_state(&mut self, raft_group_id: u64, state: &RaftLocalState) -> Result<()>;
 
     fn is_empty(&self) -> bool;
+
+    fn data_size(&self) -> usize;
```

Comment:
Is it necessary? There is only 1 write batch in one poller thread. And it is very small.